sparkles/tests/test_review.py

```diff
@@ -182,3 +184,113 @@ def test_run_aca_review_function():
         {'text': 'P2: 2.84 less than 3.0 for ER', 'category': 'critical'},
         {'text': 'ER count of 9th (8.9 for -9.9C) mag guide stars 1.91 < 3.0',
          'category': 'critical'}]
+
+
+def test_roll_outside_range():
+    kw = {'att': [-0.82389459, -0.1248412, 0.35722113, 0.42190692],
+          'date': '2019:073:21:55:30.000',
+          'detector': 'ACIS-S',
+          'dither_acq': (7.9992, 7.9992),
+          'dither_guide': (7.9992, 7.9992),
+          'focus_offset': 0.0,
+          'man_angle': 122.97035882921071,
+          'n_acq': 8,
+          'n_fid': 0,
+          'n_guide': 8,
+          'obsid': 48334.0,
+          'sim_offset': 0.0,
+          't_ccd_acq': -10.257559323423214,
+          't_ccd_guide': -10.25810835536192}
+    aca = get_aca_catalog(**kw)
+    acar = aca.get_review_table()
+    acar.get_roll_options()
+    assert Quat(kw['att']).roll <= acar.roll_info['roll_max']
+    assert Quat(kw['att']).roll >= acar.roll_info['roll_min']
+
+
+def test_calc_targ_from_aca():
+    """
+    Confirm _calc_targ_from_aca seems to do the right thing based on obsid
+    This does a bit too much processing for what should be a lightweight test.
+    """
+    obs_kwargs = KWARGS_48464
+    aca_er = get_aca_catalog(**obs_kwargs)
+    acar_er = aca_er.get_review_table()
+    q_targ = acar_er._calc_targ_from_aca(acar_er.att, 0, 0)
+    # Cheat and use the string reprs for comparison to 8 decimals
+    assert Quat(obs_kwargs['att']).__repr__() == q_targ.__repr__()
+
+    or_obs_kwargs = obs_kwargs.copy()
+    or_obs_kwargs['obsid'] = 1
+    aca_or = get_aca_catalog(**or_obs_kwargs)
+    acar_or = aca_or.get_review_table()
+    q_targ = acar_or._calc_targ_from_aca(acar_or.att, 0, 0)
+    # Assert that the the OR version *doesn't* match and is off by about 70 arcsecs yaw
+    assert Quat(obs_kwargs['att']).__repr__() != q_targ.__repr__()
+    assert abs(Quat(obs_kwargs['att']).dq(q_targ).yaw * 3600) - 69.59 < .01
+
+
+def test_get_roll_intervals():
+    """
+    Test that the behavior of get_roll_intervals is different for ORs and ERs with
+    regard to use of offsets.  They are set to arbitrary large values in the test.
+    """
+
+    # This uses the catalog at KWARGS_48464, but would really be better as a fully
```

Comment:
The roll options code doesn't yet work with synthetic catalogs.  That would be a good thing.